### DISCUSSION TITLE
ScrollTrigger on route change issues

### DISCUSSION THREAD

**Discussion by Bart-studio** [hypernym-studio/nuxt-gsap#58] [Q&A]
Hi, 

On the website I'm building with Nuxt 3, I'm using scrolltrigger to animate portions of the page.
The thing is when I navigate from page to page the animation eventually freezes...

I've seen a lot of post about this issue, and it seems I have to use 
    $ScrollTrigger.getById('idofscrolltrigger').kill()

It doesn't seems to work so I tried to console.log($ScrollTrigger.getById('uniqueId'))
And in the console I get this message : 

_ƒ t(n,r){Fi||t.register(G)||console.warn("**Please gsap.registerPlugin(ScrollTrigger)**"),zc(this),this.init(n,r)}_

Is it normal ? If not how can I register the plugin scrollTrigger ?

Below is the code I'm using.
```
const {  $gsap, $ScrollTrigger } = useNuxtApp()

function indexAnimation() {
  let tlIndex = $gsap.timeline({
      scrollTrigger: {
        id: 'agoraAgora',
        trigger: "#agoraAgora",
        start: "top bottom",
        end: "top -200",
        scrub: 1,
        // pin: true,
        // pinSpacing: true,
          markers: true,
        toggleActions: "play play reverse reverse",
      }
  });
    
  tlIndex.from("#agoraAgora .vertical-line_1", { opacity: 0, height: 0, duration: 0.25 })
          .from("#agoraAgora .pageSection__title", { opacity: 0, scale: 0.5, duration: 0.25 } , "-=0.25")
          .from("#agoraAgora .vertical-line_2", { opacity: 0, height: 0, duration: 0.25 })
          .from("#agoraAgora .content-wrapper", { opacity: 0, scale: 0.5, duration: 0.8 } , "-=0.15")
          .from("#agoraAgora .img-top-right", { opacity: 0,  x:-100, y:100, duration: 0.25 } , "+=0.25")
          .from("#agoraAgora .img-bottom-left",{ opacity: 0,  x:100, y:-100, duration: 0.25 } , "-=0.5")
          .from("#agoraAgora .vertical-line_3", { opacity: 0, height: 0 , duration: 0.25 }) 


  return tlIndex
}

onMounted(() => {
  indexAnimation()
})

onBeforeUnmount(() => {
 console.log($ScrollTrigger.getById('agoraAgora'))
  $ScrollTrigger.getById('agoraAgora').kill()
}) 
```

I've been working on it for days, so any help would be great ! 
Thank's

**Comment by bonomite** [comment]
Be sure your elements have a unique key attribute. <div :key="unique value">

**Reply by Bart-studio** [reply]
Thank you . I changed the ID and the console displays now data on my scrollTrigger, but the animation remains frozen when I navigate from page to page (in fact, the first time the page is displayed, it works as expected, but when I return to a page I've already visited, the animation is buggy). Do I have to use the refresh function or something else ?

**Comment by bonomite** [comment]
Share the repo... or share a stackblitz with the issue recreated. I've seen
this kind of issue before when routing, and it has always been a key issue.
I'd be happy to take a peek.

**Reply by Bart-studio** [reply]
I will have to test further but it seems that I have resolved my issue simply by adding a Transition component around my page content - and by doing this it looks like you don't even need to refresh or kill the scrollTrigger...

**Comment by bonomite** [comment]
Oh. Interesting. You doing this on the page level, or in a template?

**Reply by Bart-studio** [reply]
It seems to work on both levels. BUT I've tried the same thing on another site I'm building and if the page has more than one Scrolltrigger it no longer works - even if I use .kill or .refresh... I'll try to put a transition component around each element that is being animated to see what happens...

**Reply by Bart-studio** [reply]
... I tried today and nothing works anymore so back to starting point...

**Comment by bonomite** [comment]
Maybe try using refs instead of ids

**Comment by bonomite** [comment]
:(

Recreate the issue in a Stackblitz project and share it with me.
